## PROJECT DESCRIPTION

Sales analytics is the practice of generating insights from sales data, trends, and metrics to set targets and forecast future sales performance.

In this analytics project,  I explore the data to evaluate the performance of the sales team against specified goals. 
The project provides insights about the top performing and underperforming products/services, the problems in selling and market opportunities, sales forecasting, and sales activities that generate revenue.

### PROJECT QUESTIONS AND ANALYSIS

In the analysis of the project data, these questions will be answered with accompanying visualizations.

1: What was the best Year for sales? How much was earned that Year?

2: What City had the highest number of sales?

3: What was the best month for sales? How much was earned that month?

4: What products are most often sold together?

5: What product sold the most? Why do you think it sold the most?

6: What time should we display adverstisement to maximize likelihood of customer's buying product?

7: What is the probability that people will order USB-C Charging Cable?

8: What is the probability that people will orderiPhone?

9: What is the probability that people will order Google Phone?

10:What is the probability that people will order Wired Headphones?

### IMPORT LIBRARIES

In [1]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib
from matplotlib import pyplot as plt

%matplotlib inline

### IMPORT DATASETS

In [46]:
jan_data = pd.read_csv("Sales_January_2019.csv")
feb_data = pd.read_csv("Sales_February_2019.csv")
mar_data = pd.read_csv("Sales_March_2019.csv")
apr_data = pd.read_csv("Sales_April_2019.csv")
may_data = pd.read_csv("Sales_May_2019.csv")
jun_data = pd.read_csv("Sales_June_2019.csv")
jul_data = pd.read_csv("Sales_July_2019.csv")
aug_data = pd.read_csv("Sales_August_2019.csv")
sep_data = pd.read_csv("Sales_September_2019.csv")
oct_data = pd.read_csv("Sales_October_2019.csv")
nov_data = pd.read_csv("Sales_November_2019.csv")
dec_data = pd.read_csv("Sales_December_2019.csv")

## Analyze details of each of the datasets

### January product sales

In [21]:
# Checking the head of data

jan_data.head()

Order ID                   Product Quantity Ordered Price Each  \
0   141234                    iPhone                1        700   
1   141235  Lightning Charging Cable                1      14.95   
2   141236          Wired Headphones                2      11.99   
3   141237          27in FHD Monitor                1     149.99   
4   141238          Wired Headphones                1      11.99   

       Order Date                       Purchase Address  
0  01/22/19 21:25        944 Walnut St, Boston, MA 02215  
1  01/28/19 14:15       185 Maple St, Portland, OR 97035  
2  01/17/19 13:33  538 Adams St, San Francisco, CA 94016  
3  01/05/19 20:33     738 10th St, Los Angeles, CA 90001  
4  01/25/19 11:59          387 10th St, Austin, TX 73301

In [70]:
# Checking the shape of the data

jan_data.shape

(9671, 6)

There are 9671 rows and 6 columns in the dataset

In [22]:
# Checking basic info of data

jan_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9723 entries, 0 to 9722
Data columns (total 6 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   Order ID          9697 non-null   object
 1   Product           9697 non-null   object
 2   Quantity Ordered  9697 non-null   object
 3   Price Each        9697 non-null   object
 4   Order Date        9697 non-null   object
 5   Purchase Address  9697 non-null   object
dtypes: object(6)
memory usage: 455.9+ KB


In [31]:
# Checking for null values

jan_data.isna().sum()

Order ID            26
Product             26
Quantity Ordered    26
Price Each          26
Order Date          26
Purchase Address    26
dtype: int64

There are 26 instances of NaN values in the dataset. These entries will be dropped.

In [48]:
# Drop NaN values

jan_data.dropna(how="all", axis=0, inplace=True)

#### Checking for duplicated entries

In [49]:
jan_data.duplicated().any()

True

In [50]:
# Checking the number of duplicates

jan_data.duplicated().sum()

25

In [51]:
# Investigating the occurrence of duplicates

jan_data[jan_data.duplicated()]

Order ID                   Product  Quantity Ordered  Price Each  \
875     142071     AA Batteries (4-pack)                 1        3.84   
1102  Order ID                   Product  Quantity Ordered  Price Each   
1194  Order ID                   Product  Quantity Ordered  Price Each   
1897  Order ID                   Product  Quantity Ordered  Price Each   
2463  Order ID                   Product  Quantity Ordered  Price Each   
3115  Order ID                   Product  Quantity Ordered  Price Each   
3247  Order ID                   Product  Quantity Ordered  Price Each   
3612  Order ID                   Product  Quantity Ordered  Price Each   
3623  Order ID                   Product  Quantity Ordered  Price Each   
4126    145143  Lightning Charging Cable                 1       14.95   
4901  Order ID                   Product  Quantity Ordered  Price Each   
5593  Order ID                   Product  Quantity Ordered  Price Each   
5811    146765              Google Phone                 1         600   
6506  Order ID                   Product  Quantity Ordered  Price Each   
6807    147707          Wired Headphones                 1       11.99   
7544  Order ID                   Product  Quantity Ordered  Price Each   
7674  Order ID                   Product  Quantity Ordered  Price Each   
8134    148984      USB-C Charging Cable                 1       11.95   
8271  Order ID                   Product  Quantity Ordered  Price Each   
8309    149149  Lightning Charging Cable                 1       14.95   
8470    149308  Apple Airpods Headphones                 1         150   
8690    149515      USB-C Charging Cable                 1       11.95   
8923    149738      USB-C Charging Cable                 1       11.95   
9311  Order ID                   Product  Quantity Ordered  Price Each   
9427    150216          Wired Headphones                 1       11.99   

          Order Date                           Purchase Address  
875   01/17/19 23:02               131 2nd St, Boston, MA 02215  
1102      Order Date                           Purchase Address  
1194      Order Date                           Purchase Address  
1897      Order Date                           Purchase Address  
2463      Order Date                           Purchase Address  
3115      Order Date                           Purchase Address  
3247      Order Date                           Purchase Address  
3612      Order Date                           Purchase Address  
3623      Order Date                           Purchase Address  
4126  01/06/19 03:01  182 Jefferson St, San Francisco, CA 94016  
4901      Order Date                           Purchase Address  
5593      Order Date                           Purchase Address  
5811  01/21/19 11:23   918 Highland St, New York City, NY 10001  
6506      Order Date                           Purchase Address  
6807  01/04/19 16:50               883 4th St, Dallas, TX 75001  
7544      Order Date                           Purchase Address  
7674      Order Date                           Purchase Address  
8134  01/08/19 17:36              562 14th St, Boston, MA 02215  
8271      Order Date                           Purchase Address  
8309  01/12/19 12:30               180 1st St, Boston, MA 02215  
8470  01/02/19 23:07    351 Madison St, New York City, NY 10001  
8690  01/14/19 21:19         913 10th St, Los Angeles, CA 90001  
8923  01/11/19 11:22       612 West St, New York City, NY 10001  
9311      Order Date                           Purchase Address  
9427  01/21/19 09:20       691 Pine St, San Francisco, CA 94016

There are instances where "Order ID" column has non-numeric values. Drop these rows from the dataset.

In [52]:
# Drop rows that have Order ID as values in the "Order ID" column

jan_data = jan_data[jan_data["Order ID"] != "Order ID"]

In [53]:
jan_data[jan_data.duplicated()]

Order ID                   Product Quantity Ordered Price Each  \
875    142071     AA Batteries (4-pack)                1       3.84   
4126   145143  Lightning Charging Cable                1      14.95   
5811   146765              Google Phone                1        600   
6807   147707          Wired Headphones                1      11.99   
8134   148984      USB-C Charging Cable                1      11.95   
8309   149149  Lightning Charging Cable                1      14.95   
8470   149308  Apple Airpods Headphones                1        150   
8690   149515      USB-C Charging Cable                1      11.95   
8923   149738      USB-C Charging Cable                1      11.95   
9427   150216          Wired Headphones                1      11.99   

          Order Date                           Purchase Address  
875   01/17/19 23:02               131 2nd St, Boston, MA 02215  
4126  01/06/19 03:01  182 Jefferson St, San Francisco, CA 94016  
5811  01/21/19 11:23   918 Highland St, New York City, NY 10001  
6807  01/04/19 16:50               883 4th St, Dallas, TX 75001  
8134  01/08/19 17:36              562 14th St, Boston, MA 02215  
8309  01/12/19 12:30               180 1st St, Boston, MA 02215  
8470  01/02/19 23:07    351 Madison St, New York City, NY 10001  
8690  01/14/19 21:19         913 10th St, Los Angeles, CA 90001  
8923  01/11/19 11:22       612 West St, New York City, NY 10001  
9427  01/21/19 09:20       691 Pine St, San Francisco, CA 94016

#### Investigate the issue of duplicates

In [57]:
# Order ID --- 142071

jan_data[jan_data["Order ID"] == "142071"]

Order ID                Product Quantity Ordered Price Each  \
874   142071  AA Batteries (4-pack)                1       3.84   
875   142071  AA Batteries (4-pack)                1       3.84   

         Order Date              Purchase Address  
874  01/17/19 23:02  131 2nd St, Boston, MA 02215  
875  01/17/19 23:02  131 2nd St, Boston, MA 02215

In [58]:
# Order ID --- 145143

jan_data[jan_data["Order ID"] == "145143"]

Order ID                   Product Quantity Ordered Price Each  \
4125   145143  Lightning Charging Cable                1      14.95   
4126   145143  Lightning Charging Cable                1      14.95   

          Order Date                           Purchase Address  
4125  01/06/19 03:01  182 Jefferson St, San Francisco, CA 94016  
4126  01/06/19 03:01  182 Jefferson St, San Francisco, CA 94016

In [59]:
# Order ID --- 146765

jan_data[jan_data["Order ID"] == "146765"]

Order ID       Product Quantity Ordered Price Each      Order Date  \
5810   146765  Google Phone                1        600  01/21/19 11:23   
5811   146765  Google Phone                1        600  01/21/19 11:23   

                              Purchase Address  
5810  918 Highland St, New York City, NY 10001  
5811  918 Highland St, New York City, NY 10001

In [60]:
# Order ID --- 147707

jan_data[jan_data["Order ID"] == "147707"]

Order ID           Product Quantity Ordered Price Each      Order Date  \
6806   147707  Wired Headphones                1      11.99  01/04/19 16:50   
6807   147707  Wired Headphones                1      11.99  01/04/19 16:50   

                  Purchase Address  
6806  883 4th St, Dallas, TX 75001  
6807  883 4th St, Dallas, TX 75001

In [61]:
# Order ID --- 148984

jan_data[jan_data["Order ID"] == "148984"]

Order ID               Product Quantity Ordered Price Each  \
8133   148984  USB-C Charging Cable                1      11.95   
8134   148984  USB-C Charging Cable                1      11.95   

          Order Date               Purchase Address  
8133  01/08/19 17:36  562 14th St, Boston, MA 02215  
8134  01/08/19 17:36  562 14th St, Boston, MA 02215

In [62]:
# Order ID --- 149149

jan_data[jan_data["Order ID"] == "149149"]

Order ID                   Product Quantity Ordered Price Each  \
8308   149149  Lightning Charging Cable                1      14.95   
8309   149149  Lightning Charging Cable                1      14.95   

          Order Date              Purchase Address  
8308  01/12/19 12:30  180 1st St, Boston, MA 02215  
8309  01/12/19 12:30  180 1st St, Boston, MA 02215

In [63]:
# Order ID --- 149308

jan_data[jan_data["Order ID"] == "149308"]

Order ID                   Product Quantity Ordered Price Each  \
8469   149308  Apple Airpods Headphones                1        150   
8470   149308  Apple Airpods Headphones                1        150   

          Order Date                         Purchase Address  
8469  01/02/19 23:07  351 Madison St, New York City, NY 10001  
8470  01/02/19 23:07  351 Madison St, New York City, NY 10001

In [64]:
# Order ID --- 149515

jan_data[jan_data["Order ID"] == "149515"]

Order ID               Product Quantity Ordered Price Each  \
8689   149515  USB-C Charging Cable                1      11.95   
8690   149515  USB-C Charging Cable                1      11.95   

          Order Date                    Purchase Address  
8689  01/14/19 21:19  913 10th St, Los Angeles, CA 90001  
8690  01/14/19 21:19  913 10th St, Los Angeles, CA 90001

In [65]:
# Order ID --- 149738

jan_data[jan_data["Order ID"] == "149738"]

Order ID               Product Quantity Ordered Price Each  \
8922   149738  USB-C Charging Cable                1      11.95   
8923   149738  USB-C Charging Cable                1      11.95   

          Order Date                      Purchase Address  
8922  01/11/19 11:22  612 West St, New York City, NY 10001  
8923  01/11/19 11:22  612 West St, New York City, NY 10001

In [66]:
# Order ID --- 150216

jan_data[jan_data["Order ID"] == "150216"]

Order ID           Product Quantity Ordered Price Each      Order Date  \
9426   150216  Wired Headphones                1      11.99  01/21/19 09:20   
9427   150216  Wired Headphones                1      11.99  01/21/19 09:20   

                          Purchase Address  
9426  691 Pine St, San Francisco, CA 94016  
9427  691 Pine St, San Francisco, CA 94016

From the duplicates investigations, it can be realised that each duplicated entries had same values. One instance of each duplicated entry will be dropped.

In [67]:
# Drop duplicated entries

jan_data.drop_duplicates(subset=None, keep="first", inplace=True)

### February product sales

In [71]:
# Checking the head of the data

feb_data.head()

Order ID                   Product Quantity Ordered Price Each  \
0   150502                    iPhone                1        700   
1   150503     AA Batteries (4-pack)                1       3.84   
2   150504    27in 4K Gaming Monitor                1     389.99   
3   150505  Lightning Charging Cable                1      14.95   
4   150506     AA Batteries (4-pack)                2       3.84   

       Order Date                     Purchase Address  
0  02/18/19 01:35    866 Spruce St, Portland, ME 04101  
1  02/13/19 07:24  18 13th St, San Francisco, CA 94016  
2  02/18/19 09:46   52 6th St, New York City, NY 10001  
3  02/02/19 16:47     129 Cherry St, Atlanta, GA 30301  
4  02/28/19 20:32    548 Lincoln St, Seattle, WA 98101

In [72]:
# Checking the shape of the data

feb_data.shape

(12036, 6)

There are 12036 rows and 6 columns in the data

In [73]:
# Checking the basic info of the data

feb_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12036 entries, 0 to 12035
Data columns (total 6 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   Order ID          12004 non-null  object
 1   Product           12004 non-null  object
 2   Quantity Ordered  12004 non-null  object
 3   Price Each        12004 non-null  object
 4   Order Date        12004 non-null  object
 5   Purchase Address  12004 non-null  object
dtypes: object(6)
memory usage: 564.3+ KB


In [74]:
# Checking for the null values in the data

feb_data.isna().sum()

Order ID            32
Product             32
Quantity Ordered    32
Price Each          32
Order Date          32
Purchase Address    32
dtype: int64

There are 32 instances of NaN values in the data

In [75]:
# Drop NaN values from the data

feb_data.dropna(how="all", axis=0, inplace=True)

In [77]:
# Checking for duplicates in the data

feb_data.duplicated().any()

True

In [78]:
# Checking the number of duplicates

feb_data.duplicated().sum()

35

There are 35 instances of duplicated entries in the data

In [79]:
# Investigating the occurrence of duplicates

feb_data[feb_data.duplicated()]

Order ID                     Product  Quantity Ordered  Price Each  \
432      150917    Lightning Charging Cable                 1       14.95   
442      150925                      iPhone                 1         700   
461      150943        USB-C Charging Cable                 1       11.95   
548      151024            Wired Headphones                 1       11.99   
1164     151616        USB-C Charging Cable                 1       11.95   
1224     151673            Wired Headphones                 1       11.99   
1417     151856        USB-C Charging Cable                 1       11.95   
1904   Order ID                     Product  Quantity Ordered  Price Each   
1918     152330  Bose SoundSport Headphones                 1       99.99   
2050   Order ID                     Product  Quantity Ordered  Price Each   
2225   Order ID                     Product  Quantity Ordered  Price Each   
2937     153304            Wired Headphones                 1       11.99   
2949     153315            Wired Headphones                 1       11.99   
3091   Order ID                     Product  Quantity Ordered  Price Each   
3128   Order ID                     Product  Quantity Ordered  Price Each   
3142   Order ID                     Product  Quantity Ordered  Price Each   
3360   Order ID                     Product  Quantity Ordered  Price Each   
3426   Order ID                     Product  Quantity Ordered  Price Each   
3831   Order ID                     Product  Quantity Ordered  Price Each   
4013   Order ID                     Product  Quantity Ordered  Price Each   
4070   Order ID                     Product  Quantity Ordered  Price Each   
4446     154737    Lightning Charging Cable                 1       14.95   
4457     154747      27in 4K Gaming Monitor                 1      389.99   
4518   Order ID                     Product  Quantity Ordered  Price Each   
5315   Order ID                     Product  Quantity Ordered  Price Each   
5450     155697       AA Batteries (4-pack)                 1        3.84   
5880     156109  Bose SoundSport Headphones                 1       99.99   
6025     156247      AAA Batteries (4-pack)                 1        2.99   
6099   Order ID                     Product  Quantity Ordered  Price Each   
6957   Order ID                     Product  Quantity Ordered  Price Each   
7301   Order ID                     Product  Quantity Ordered  Price Each   
8111     158236       AA Batteries (4-pack)                 1        3.84   
8734     158841      34in Ultrawide Monitor                 1      379.99   
11528  Order ID                     Product  Quantity Ordered  Price Each   
11574    161567    Apple Airpods Headphones                 1         150   

           Order Date                         Purchase Address  
432    02/06/19 16:07            111 10th St, Austin, TX 73301  
442    02/07/19 17:43             784 Elm St, Boston, MA 02215  
461    02/06/19 19:13             759 1st St, Austin, TX 73301  
548    02/19/19 08:39           35 Pine St, Portland, OR 97035  
1164   02/25/19 19:29          666 Meadow St, Boston, MA 02215  
1224   02/10/19 21:52          504 Center St, Dallas, TX 75001  
1417   02/06/19 12:11  475 Jackson St, San Francisco, CA 94016  
1904       Order Date                         Purchase Address  
1918   02/25/19 18:53    827 Dogwood St, Los Angeles, CA 90001  
2050       Order Date                         Purchase Address  
2225       Order Date                         Purchase Address  
2937   02/12/19 20:08           74 Meadow St, Austin, TX 73301  
2949   02/13/19 14:47      953 Jefferson St, Atlanta, GA 30301  
3091       Order Date                         Purchase Address  
3128       Order Date                         Purchase Address  
3142       Order Date                         Purchase Address  
3360       Order Date                         Purchase Address  
3426       Order Date                         Purchase Address  

There are instances where "Order ID" column has non-numeric values. Drop these rows from the dataset.

In [81]:
# Drop rows that have Order ID as values in the "Order ID" column

feb_data = feb_data[feb_data["Order ID"] != "Order ID"]

In [82]:
feb_data[feb_data.duplicated()]

Order ID                     Product Quantity Ordered Price Each  \
432     150917    Lightning Charging Cable                1      14.95   
442     150925                      iPhone                1        700   
461     150943        USB-C Charging Cable                1      11.95   
548     151024            Wired Headphones                1      11.99   
1164    151616        USB-C Charging Cable                1      11.95   
1224    151673            Wired Headphones                1      11.99   
1417    151856        USB-C Charging Cable                1      11.95   
1918    152330  Bose SoundSport Headphones                1      99.99   
2937    153304            Wired Headphones                1      11.99   
2949    153315            Wired Headphones                1      11.99   
4446    154737    Lightning Charging Cable                1      14.95   
4457    154747      27in 4K Gaming Monitor                1     389.99   
5450    155697       AA Batteries (4-pack)                1       3.84   
5880    156109  Bose SoundSport Headphones                1      99.99   
6025    156247      AAA Batteries (4-pack)                1       2.99   
8111    158236       AA Batteries (4-pack)                1       3.84   
8734    158841      34in Ultrawide Monitor                1     379.99   
11574   161567    Apple Airpods Headphones                1        150   

           Order Date                         Purchase Address  
432    02/06/19 16:07            111 10th St, Austin, TX 73301  
442    02/07/19 17:43             784 Elm St, Boston, MA 02215  
461    02/06/19 19:13             759 1st St, Austin, TX 73301  
548    02/19/19 08:39           35 Pine St, Portland, OR 97035  
1164   02/25/19 19:29          666 Meadow St, Boston, MA 02215  
1224   02/10/19 21:52          504 Center St, Dallas, TX 75001  
1417   02/06/19 12:11  475 Jackson St, San Francisco, CA 94016  
1918   02/25/19 18:53    827 Dogwood St, Los Angeles, CA 90001  
2937   02/12/19 20:08           74 Meadow St, Austin, TX 73301  
2949   02/13/19 14:47      953 Jefferson St, Atlanta, GA 30301  
4446   02/01/19 17:59     513 10th St, San Francisco, CA 94016  
4457   02/01/19 22:46           367 Cedar St, Austin, TX 73301  
5450   02/13/19 15:17          961 Spruce St, Boston, MA 02215  
5880   02/18/19 09:18         450 Jackson St, Boston, MA 02215  
6025   02/09/19 07:29    511 Dogwood St, Los Angeles, CA 90001  
8111   02/19/19 09:49     319 West St, San Francisco, CA 94016  
8734   02/01/19 23:16          786 Willow St, Boston, MA 02215  
11574  02/10/19 11:42   413 Walnut St, San Francisco, CA 94016

#### Investigating the issue of duplicated entries